### ISSUE TITLE
What is the rationale behind 100 chars per line for the checkstyle? 

### ISSUE THREAD

**Issue by yuehernkang** [nus-cs2030/2122-s1#338]
# Summary
The last stage of the project is the checkstyle stage which limits one line to 100 characters

## Description 
Is this standard practice in the industry? Sometimes it may get even more messy when we put code in the next line rather than in 1 line 

## Screenshots (if any): 
Insert Images here if necessary

**Comment by RussellDash332** [comment]
Not sure why, but my feeling is for convenience since not all text editors have the text wrapping feature, so might as well manually wrap it.

**Comment by seasonedlasagna** [comment]
It is to improve readability of code since not everyone working with your code would have widescreen monitors (assuming you're working in a large software engineering team / need to collaborate). Plus, it's easy to lose your train of thought when the line gets too long. 

I think it's also Google's Java coding style (they started off with a limit of 80 chars per line then increased it with the advent of bigger monitor size). https://google.github.io/styleguide/javaguide.html

**Comment by albertsutz** [comment]
One possible reason is to increase readability and to practice coding efficiently. If you can make your code shorter, why not?

^^ just a thought.

**Comment by yuehernkang** [comment]
> It is to improve readability of code since not everyone working with your code would have widescreen monitors (assuming you're working in a large software engineering team / need to collaborate). Plus, it's easy to lose your train of thought when the line gets too long.
> 
> I think it's also Google's Java coding style (they started off with a limit of 80 chars per line then increased it with the advent of bigger monitor size). https://google.github.io/styleguide/javaguide.html

This is interesting. Thanks for sharing !

**Comment by Devvynneo** [comment]
I think its to help the structure and cleanliness of the code
Imagine if everything is 100 + lines, it would be everything difficult to read and edit

**Comment by va5hi** [comment]
yeah i agree as well i think if sort of forces us to avoid hard coding and make our codes more succinct as well

**Comment by kiwizop** [comment]
Agree, imagine having to scroll left and right if there are too many words in a line
